src/internet_identity/src/anchor_management.rs

```diff
@@ -109,6 +109,36 @@ pub fn update(user_number: AnchorNumber, device_key: DeviceKey, device_data: Dev
     );
 }
 
+pub fn replace(anchor_number: AnchorNumber, device_key: DeviceKey, device_data: DeviceData) {
+    let mut anchor = state::anchor(anchor_number);
+    // must be called before the first await because it requires caller()
+    trap_if_not_authenticated(&anchor);
+
+    anchor.remove_device(&device_key).unwrap_or_else(|err| {
+        trap(&format!(
+            "failed to replace device of anchor {}: {}",
+            anchor_number, err
+        ))
+    });
+    let new_device = Device::from(device_data);
+    anchor.add_device(new_device.clone()).unwrap_or_else(|err| {
+        trap(&format!(
+            "failed to replace device of anchor {}: {}",
+            anchor_number, err
+        ))
+    });
+    write_anchor(anchor_number, anchor);
+    archive_operation(
+        anchor_number,
+        caller(),
+        Operation::ReplaceDevice {
+            device: device_key,
```

Comment:
note: I didn't immediately realize this was the "old" device. Since the other field is `new_device` is expected `old_device` or something